src/QueryLanguage/Processor/Doctrine/Dbal/Processor.php

```diff
@@ -0,0 +1,160 @@
+        if ($result instanceof FormInterface) {
+            return $result;
+        }
+
+        $this->attachToQueryBuilder($result->filters);
+        $pageSize = $this->options['default_page_size'] ?? $result->limit;
+
+        if (null !== $result->skip) {
+            $this->queryBuilder->setFirstResult($result->skip);
+        }
+
+        if (null !== $result->ordering) {
+            if ($this->options['continuation_token']) {
+                $iterator = new PagerIterator($this->queryBuilder, $this->parseOrderings($result->ordering));
+                $iterator->setToken($result->pageToken);
+                if (null !== $pageSize) {
+                    $iterator->setPageSize($pageSize);
+                }
+
+                return $iterator;
+            }
+
+            $sql = $this->queryBuilder->getSQL();
+            $direction = $result->ordering->getDirection();
+            $fieldName = $this->columns[$result->ordering->getField()]->fieldName;
+
+            $this->queryBuilder = $this->queryBuilder->getConnection()
+                ->createQueryBuilder()
+                ->select('*')
+                ->from("($sql)", 'x')
+                ->setFirstResult($result->skip ?? 0)
+                ->orderBy($fieldName, $direction)
+            ;
+        }
+
+        if (null !== $pageSize) {
+            $this->queryBuilder->setMaxResults($pageSize);
+        }
+
+        return new RowIterator($this->queryBuilder);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function createColumn(string $fieldName): ColumnInterface
+    {
+        $tableName = null;
+        if (false !== \strpos($fieldName, '.')) {
+            [$tableName, $fieldName] = \explode('.', $fieldName);
+        }
+
+        return new Column($fieldName, $tableName);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getIdentifierFieldNames(): array
+    {
+        return $this->identifierFields;
+    }
+
+    /**
+     * Parses the ordering expression for continuation token.
+     *
+     * @param OrderExpression $ordering
+     *
+     * @return array
+     */
+    protected function parseOrderings(OrderExpression $ordering): array
+    {
+        $checksumColumn = $this->options['continuation_token']['checksum_field'] ?? $this->getIdentifierFieldNames()[0] ?? null;
+        $column = $this->columns[$ordering->getField()];
+
+        if (null === $checksumColumn) {
+            foreach ($this->columns as $column) {
+                if ($checksumColumn === $column) {
+                    continue;
+                }
+
+                $checksumColumn = $column;
+                break;
+            }
+        }
+
+        $checksumField = $checksumColumn instanceof Column ? $checksumColumn->fieldName : $checksumColumn;
+        $fieldName = $column->fieldName;
+        $direction = $ordering->getDirection();
+
+        return [
+            $fieldName => $direction,
+            $checksumField => 'ASC',
```

Comment:
Do we have a constant for `ASC`? I don't remember 🤔

Reply:
Nope

Reply:
👍